src/main/java/com/prykhodko/shop/service/impl/ConfirmationCodeServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.prykhodko.shop.service.impl;
+
+
+import com.prykhodko.shop.dao.ConfirmationCodeDao;
+import com.prykhodko.shop.model.ConfirmationCode;
+import com.prykhodko.shop.model.User;
+import com.prykhodko.shop.service.ConfirmationCodeService;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional
+public class ConfirmationCodeServiceImpl implements ConfirmationCodeService {
+
+    private static final Logger logger = Logger.getLogger(ConfirmationCodeServiceImpl.class);
+
+    private final ConfirmationCodeDao confirmationCodeDao;
+
+    @Autowired
+    public ConfirmationCodeServiceImpl(ConfirmationCodeDao confirmationCodeDao) {
+        this.confirmationCodeDao = confirmationCodeDao;
+    }
+
+    @Override
+    @Transactional
```

Comment:
we use Transactional over the class, so we don't need it here

Reply:
fixed